### PR TITLE
Add functionality to restrict access to next module based on passing previous module

### LINKED ISSUE #597: Functionality to restrict access to next module based on passing previous module

### PR DESCRIPTION
**Pull request by ankitjavalkar**
@prathamesh920 @adityacp @CruiseDevice Please review, will fix tests on an ongoing basis and keep updating the PR

Closes #597

### CODE REVIEW

File: yaksh/models.py [hunk 1]
```diff
@@ -539,6 +539,18 @@ def get_answerpaper_status(self, user, course):
...
         return status
 
+    def get_answerpaper_passing_status(self, user, course):
+        try:
+            qp = self.questionpaper_set.get().id
+        except QuestionPaper.DoesNotExist:
+            qp = None
+        ans_ppr = AnswerPaper.objects.filter(
+            user=user, course=course, question_paper=qp
+        ).order_by("-attempt_number")
```
Comment (prathamesh920) [inline yaksh/models.py]: Should this be order by percentage?
As there can be a case that in 2 attempts, a user passed in the first and failed in the second!
What are your thoughts?

File: yaksh/models.py [hunk 2]
```diff
@@ -698,6 +714,33 @@ def is_prerequisite_passed(self, user, course):
                 success = False
         return success
 
+    def get_passing_status(self, user, course):
+        course_status = CourseStatus.objects.filter(user=user, course=course)
+        state = "failed"
```
Comment (prathamesh920) [inline yaksh/models.py]: This variable is not used.

File: yaksh/models.py [hunk 3]
```diff
@@ -698,6 +714,33 @@ def is_prerequisite_passed(self, user, course):
...
+        if course_status.exists():
+            ordered_units = self.learning_unit.filter(type='quiz').order_by("order")
+
+        status_list = [unit.quiz.get_answerpaper_passing_status(user, course)
```
Comment (prathamesh920) [inline yaksh/models.py]: Can we have a name like ```all_status``` or  may be ```statuses```?

**Comment by prathamesh920** [comment]
Also fix PEP8

File: yaksh/models.py (hunk 3 above)
Comment (ankitjavalkar) [reply yaksh/models.py]: Done

File: yaksh/models.py (hunk 2 above)
Comment (ankitjavalkar) [reply yaksh/models.py]: Removed

**Comment by ankitjavalkar** [comment]
Fixed PEP8

**Comment by ankitjavalkar** [comment]
@prathamesh920 Please check. PEP8 issues have been fixed, but PEP8 issue comment is now outdated.

**Pull request merged**